Engine/Component/Component.h

```diff
@@ -23,13 +23,33 @@ class Component
 	Component(GameObject * owner, ComponentType componentType) : owner(owner), type(componentType), UUID(pcg32_random()) {};
 	virtual ~Component() = default;
 
-	virtual void Enable() { active = true; };
-	virtual void Disable() { active = false; };
+	//Copy and move
+	Component(const Component& component_to_copy) = default;
+	Component(Component&& component_to_move) = default;
```

Comment:
First time seeing the double address operator.

Reply:
Double adress indicates move constructor, they are quite usefull to speed up execution time and saving memory

Reply:
It is good practice to implement the copy constructor and the move constructor.